apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/PreAdjFissClaimResponseTransformerV2.java

```diff
@@ -0,0 +1,150 @@
+package gov.cms.bfd.server.war.r4.providers;
+
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.Timer;
+import com.newrelic.api.agent.Trace;
+import gov.cms.bfd.model.rda.PreAdjFissClaim;
+import gov.cms.bfd.sharedutils.exceptions.BadCodeMonkeyException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.hl7.fhir.r4.model.ClaimResponse;
+import org.hl7.fhir.r4.model.CodeableConcept;
+import org.hl7.fhir.r4.model.Coding;
+import org.hl7.fhir.r4.model.Extension;
+import org.hl7.fhir.r4.model.Identifier;
+import org.hl7.fhir.r4.model.Meta;
+import org.hl7.fhir.r4.model.Reference;
+
+/** Transforms FISS/MCS instances into FHIR {@link ClaimResponse} resources. */
+public class PreAdjFissClaimResponseTransformerV2 {
+
+  private static final String METRIC_NAME =
+      MetricRegistry.name(PreAdjFissClaimResponseTransformerV2.class.getSimpleName(), "transform");
+
+  private static final Map<Character, String> STATUS_TEXT;
+
+  static {
+    STATUS_TEXT = new HashMap<>();
+    STATUS_TEXT.put('a', "Active");
+    STATUS_TEXT.put('s', "Suspend");
+    STATUS_TEXT.put('p', "Paid");
+    STATUS_TEXT.put('d', "Denied");
+    STATUS_TEXT.put('i', "Inactive");
+    STATUS_TEXT.put('r', "Reject");
+    STATUS_TEXT.put('t', "RTP");
+    STATUS_TEXT.put('m', "Move");
+  }
+
+  private static final Map<Character, ClaimResponse.RemittanceOutcome> STATUS_TO_OUTCOME;
+
+  static {
+    STATUS_TO_OUTCOME = new HashMap<>();
+    STATUS_TO_OUTCOME.put(' ', ClaimResponse.RemittanceOutcome.QUEUED);
+    STATUS_TO_OUTCOME.put('a', ClaimResponse.RemittanceOutcome.QUEUED);
+    STATUS_TO_OUTCOME.put('s', ClaimResponse.RemittanceOutcome.PARTIAL);
+    STATUS_TO_OUTCOME.put('p', ClaimResponse.RemittanceOutcome.COMPLETE);
+    STATUS_TO_OUTCOME.put('d', ClaimResponse.RemittanceOutcome.ERROR);
+    STATUS_TO_OUTCOME.put('i', ClaimResponse.RemittanceOutcome.PARTIAL);
+    STATUS_TO_OUTCOME.put('r', ClaimResponse.RemittanceOutcome.ERROR);
+    STATUS_TO_OUTCOME.put('t', ClaimResponse.RemittanceOutcome.ERROR);
+    STATUS_TO_OUTCOME.put('m', ClaimResponse.RemittanceOutcome.PARTIAL);
+  }
```

Comment:
Assuming this is meant to be immutable this should be a Guava ImmutableMap rather than a HashMap.

Reply:
kyle. BFD has been trying to use enums especially when they are related to Codable logic that requires a code, a display text, and a system url. We have FHIR-CARIN types like:
gov.cms.bfd.server.war.commons.carin.C4BBAdjudication.java

And we have some that don't fit into FHIR-CARIN such as:
gov.cms.bfd.server.war.commons.RaceCategory.java

Reply:
@mackec1 thanks Chris this is really helpful.  When possible we of course want to use existing carin types, however in this case I believe [CC4BBAdjudication](http://hl7.org/fhir/us/carin-bb/STU1/ValueSet-C4BBAdjudication.html) represents "the amount fields submitted for adjudication".   Our [outcome field](http://hl7.org/fhir/valueset-remittance-outcome.html) is meant to track the "state of the claim within the adjudication process".

Is this the field that you are using for [EOB.outcome](http://hl7.org/fhir/explanationofbenefit-definitions.html#ExplanationOfBenefit.outcome)?